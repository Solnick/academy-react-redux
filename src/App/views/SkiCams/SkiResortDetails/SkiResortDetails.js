import React from 'react';
import PropTypes from 'prop-types';
import getCurrentFormattedDate from '../../../utils/getCurrentFormattedDate';
import Tile from '../../../components/Tile/Tile';
import styles from './skiResortDetails.module.scss';

const SkiResortDetails = (props) => {
  const { name: skiResortName, cams } = props;
  const camsArray = Object.values(cams).slice(0, 2);

  return (
    <Tile>
      <div className={styles.container}>
        <span className={styles.titleContainer}>
          <span>{getCurrentFormattedDate()}</span>
          <h2>{skiResortName}</h2>
        </span>
        {
          camsArray.map(({ url, name }) => (
            <img
              src={url}
              key={name}
              alt={name}
            />
          ))
        }
      </div>
    </Tile>
  );
};

SkiResortDetails.propTypes = {
  name: PropTypes.string.isRequired,
  cams: PropTypes.object.isRequired,
};

export default SkiResortDetails;
