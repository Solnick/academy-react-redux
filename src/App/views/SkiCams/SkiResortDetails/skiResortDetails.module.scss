@import '../../../style/variables';
@import '../../../style/breakpoints';

.container {
  display: flex;
  flex-direction: column;
  .titleContainer {
    display: flex;
    flex-direction: column;
    h2 {
      display: flex;
      justify-content: center;
      width: 100%;
      font-size: 1.5rem;
      margin-bottom: 15px;
    }
    span {
      display: flex;
      align-self: baseline;
      justify-content: flex-end;
      width: 100%;
      padding: 8px;
      color: lighten($font-color, 30);
      font-size: 0.85rem;
      font-weight: 500;
    }
  }
  img {
    max-height: 290px;
    height: auto;
    object-fit: scale-down;
    background-color: black;
  }
}
