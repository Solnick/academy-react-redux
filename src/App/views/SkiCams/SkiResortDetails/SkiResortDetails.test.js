import React from 'react';
import { shallow } from 'enzyme';
import SkiResortDetails from './SkiResortDetails';

describe('SkiResortDetails component', () => {
  it('should render correctly', () => {
    const props = {
      name: 'Castelluccio di Norcia',
      cams: {
        194: {
          name: 'Pian Perduto alt. 1452 mt',
          url: 'http://www.umbriameteo.com/web1a/images/castelluccio_pdah.jpg',
        },
        195: {
          name: "Vista sull'abitato di Castelluccio",
          url: 'http://www.umbriameteo.com/web1/images/lastpdah.jpg',
        },
      },
    };
    const skiResortDetailsInstance = shallow(<SkiResortDetails {...props} />);
    expect(skiResortDetailsInstance).toMatchSnapshot();
  });
});
