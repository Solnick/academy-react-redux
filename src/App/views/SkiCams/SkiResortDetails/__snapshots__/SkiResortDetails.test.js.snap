// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkiResortDetails component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SkiResortDetails
    cams={
      Object {
        "194": Object {
          "name": "Pian Perduto alt. 1452 mt",
          "url": "http://www.umbriameteo.com/web1a/images/castelluccio_pdah.jpg",
        },
        "195": Object {
          "name": "Vista sull'abitato di Castelluccio",
          "url": "http://www.umbriameteo.com/web1/images/lastpdah.jpg",
        },
      }
    }
    name="Castelluccio di Norcia"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "center": false,
      "children": <div
        className="container"
      >
        <span
          className="titleContainer"
        >
          <span>
            14-04-2019
          </span>
          <h2>
            Castelluccio di Norcia
          </h2>
        </span>
        <img
          alt="Pian Perduto alt. 1452 mt"
          src="http://www.umbriameteo.com/web1a/images/castelluccio_pdah.jpg"
        />
        <img
          alt="Vista sull'abitato di Castelluccio"
          src="http://www.umbriameteo.com/web1/images/lastpdah.jpg"
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="titleContainer"
          >
            <span>
              14-04-2019
            </span>
            <h2>
              Castelluccio di Norcia
            </h2>
          </span>,
          Array [
            <img
              alt="Pian Perduto alt. 1452 mt"
              src="http://www.umbriameteo.com/web1a/images/castelluccio_pdah.jpg"
            />,
            <img
              alt="Vista sull'abitato di Castelluccio"
              src="http://www.umbriameteo.com/web1/images/lastpdah.jpg"
            />,
          ],
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                14-04-2019
              </span>,
              <h2>
                Castelluccio di Norcia
              </h2>,
            ],
            "className": "titleContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "14-04-2019",
              },
              "ref": null,
              "rendered": "14-04-2019",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Castelluccio di Norcia",
              },
              "ref": null,
              "rendered": "Castelluccio di Norcia",
              "type": "h2",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "Pian Perduto alt. 1452 mt",
          "nodeType": "host",
          "props": Object {
            "alt": "Pian Perduto alt. 1452 mt",
            "src": "http://www.umbriameteo.com/web1a/images/castelluccio_pdah.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": "Vista sull'abitato di Castelluccio",
          "nodeType": "host",
          "props": Object {
            "alt": "Vista sull'abitato di Castelluccio",
            "src": "http://www.umbriameteo.com/web1/images/lastpdah.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "center": false,
        "children": <div
          className="container"
        >
          <span
            className="titleContainer"
          >
            <span>
              14-04-2019
            </span>
            <h2>
              Castelluccio di Norcia
            </h2>
          </span>
          <img
            alt="Pian Perduto alt. 1452 mt"
            src="http://www.umbriameteo.com/web1a/images/castelluccio_pdah.jpg"
          />
          <img
            alt="Vista sull'abitato di Castelluccio"
            src="http://www.umbriameteo.com/web1/images/lastpdah.jpg"
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="titleContainer"
            >
              <span>
                14-04-2019
              </span>
              <h2>
                Castelluccio di Norcia
              </h2>
            </span>,
            Array [
              <img
                alt="Pian Perduto alt. 1452 mt"
                src="http://www.umbriameteo.com/web1a/images/castelluccio_pdah.jpg"
              />,
              <img
                alt="Vista sull'abitato di Castelluccio"
                src="http://www.umbriameteo.com/web1/images/lastpdah.jpg"
              />,
            ],
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  14-04-2019
                </span>,
                <h2>
                  Castelluccio di Norcia
                </h2>,
              ],
              "className": "titleContainer",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "14-04-2019",
                },
                "ref": null,
                "rendered": "14-04-2019",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Castelluccio di Norcia",
                },
                "ref": null,
                "rendered": "Castelluccio di Norcia",
                "type": "h2",
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "Pian Perduto alt. 1452 mt",
            "nodeType": "host",
            "props": Object {
              "alt": "Pian Perduto alt. 1452 mt",
              "src": "http://www.umbriameteo.com/web1a/images/castelluccio_pdah.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": "Vista sull'abitato di Castelluccio",
            "nodeType": "host",
            "props": Object {
              "alt": "Vista sull'abitato di Castelluccio",
              "src": "http://www.umbriameteo.com/web1/images/lastpdah.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
